File: python/easy/valid_parentheses.py
def isValid(s):
    x = []

    for i in range(0, len(s)):
        if s[i] == '(' or s[i] == '[' or s[i] == '{':
            x.append(s[i])
        else:
            if len(x) != 0:
                if (s[i] == ')' and x[-1] != '(') or (s[i] == ']' and x[-1] != '[') or (s[i] == '}' and x[-1] != '{'):
                    return False
                else:
                    x.pop()
            else:
                return False

    if len(x) == 0:
        return True
    else:
        return False



s = "{[]}"
isValid(s)

File: javascript/easy/valid_parentheses.js
// 20. Valid Parentheses
// https://leetcode.com/problems/valid-parentheses/submissions/

var isValid = function(s) {
    var x = [];
    
    for (var i = 0; i < s.length; i++) {
        if (s[i] == '(' || s[i] == '[' || s[i] == '{') {
            x.push(s[i]);
            }
        else {
            if (!(x.length == 0)) {
                if ( s[i] == ')' && (!(x[x.length - 1] == '(')) || s[i] == ']' && (!(x[x.length - 1] == '[')) || s[i] == '}' && (!(x[x.length - 1] == '{')) ) {
                    return false;
                }
                else {
                    x.pop();
                }
            }
            else {
                return false;
            }
        }
    }
    if (x.length == 0) {
        return true;
    }
    else {
        return false;
    }
};